pkg/controller/operators/catalog/operator.go

```diff
@@ -1110,13 +1129,16 @@ func (o *Operator) ensureSubscriptionInstallPlanState(logger *logrus.Entry, sub
 	out.Status.InstallPlanRef = ref
 	out.Status.Install = v1alpha1.NewInstallPlanReference(ref)
 	out.Status.State = v1alpha1.SubscriptionStateUpgradePending
+	if failForwardEnabled && ip.Status.Phase == v1alpha1.InstallPlanPhaseFailed {
+		out.Status.State = v1alpha1.SubscriptionStateFailed
```

Comment:
should we add another conditional check for `&& out.Status.State != v1lpha1.SubscriptionStateFailed`? so we don't continuously attempt to update the status if it's already set? Or does the underlying update mechanism handle that case for us?

Reply:
In existing behavior the `out.Status.State` is always set to `v1alpha1.SubscriptionStateUpgradePending` and I haven't seen any indication of hotlooping with my tests where the conditional is met, so I think we're safe here.